pennylane/_qubit_device.py

```diff
@@ -1425,7 +1430,7 @@ def circuit(x):
         batched_ndims = 2
         shape = samples.shape
 
-        if mp.obs is None and mp.mv is None:
+        if mp.obs is None and not isinstance(mp.mv, MeasurementValue):
```

Comment:
It shouldn't be (at least mathematically) be treated as an observable. A list of `MeasurementValue`s will be processed the same way wires are.

Reply:
wondering if maybe we should consider something like `MeasurementProcess.has_observable` now. the measurement API is starting to show signs of weakness at doing its job throughout the codebase, so it might need a more proper upgrade to account for something other than observable vs. wires (in other words, mid-circuit measurements)

Reply:
I can include something like that in the measurement function singledispatch PR. We'll be creating a stronger separation between observables, wires, and measurement values in that PR, so it seems fitting.